### PR TITLE
Centralise `MapInterceptorAdaptor` in tests to avoid boilerplate code

### PR DESCRIPTION
**Pull request by JackPGreen**
Fixes https://github.com/hazelcast/hazelcast/issues/25472

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/client/test/IdentifiedDataSerializableFactory.java [hunk 1]
```diff
@@ -402,28 +404,6 @@ public Object interceptGet(Object value) {
             return prefix + val;
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/client/test/IdentifiedDataSerializableFactory.java:404]: for consistency, don't you need to call super.interceptGet here?

File: hazelcast/src/test/java/com/hazelcast/map/InterceptorTest.java [hunk 2]
```diff
@@ -341,6 +341,7 @@ public Iterable<Integer> loadAllKeys() {
     }
 
     static class EntryPutProcessor implements EntryProcessor<Integer, String, String> {
+        private static final long serialVersionUID = 1L;
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/map/InterceptorTest.java:344]: this seems like an unrelated change to me.

File: hazelcast/src/test/java/com/hazelcast/map/InterceptorTest.java [hunk 3]
```diff
@@ -382,37 +383,8 @@ String getLoadedValue() {
...
-            return removedValue;
-        }
-
-        @Override
-        public void afterRemove(Object value) {
-        }
-    }
-
     public static class SimpleInterceptor extends MapInterceptorAdaptor {
+        private static final long serialVersionUID = 1L;
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/map/InterceptorTest.java:387]: In overloaded methods of these classes, you don't call super.methodName() but you do for others. Let's pick one and use it for all.

**Review by srknzl — APPROVED** [review]
ok with a nit

File: hazelcast/src/test/java/com/hazelcast/map/InterceptorTest.java (hunk 2 above)
Comment (JackPGreen) [reply hazelcast/src/test/java/com/hazelcast/map/InterceptorTest.java]: c2772b2416f0412b1a05a0d212ec295bcafe06d9
